pkg/user/user_test.go

```diff
@@ -276,3 +276,73 @@ func TestUser_UpdateName(t *testing.T) {
 	u.UpdateName("xxx")
 	assert.Equal(t, "xxx", u.Name())
 }
+
+func TestUser_MatchPassword(t *testing.T) {
+	encodedPass, _ := encodePassword("test")
+	type args struct {
+		pass string
+	}
+	tests := []struct {
+		name     string
+		password []byte
+		args     args
+		want     bool
+		wantErr  bool
+	}{
+		{
+			name:     "passwords should match",
+			password: encodedPass,
+			args: args{
+				pass: "test",
+			},
+			want:    true,
+			wantErr: false,
+		},
+		{
+			name:     "passwords shouldn't match",
+			password: encodedPass,
+			args: args{
+				pass: "xxx",
+			},
+			want:    false,
+			wantErr: false,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(tt *testing.T) {
+			u := &User{
+				password: tc.password,
+			}
+			got, err := u.MatchPassword(tc.args.pass)
+			assert.Equal(tt, tc.want, got)
+			assert.Equal(tt, tc.wantErr, err != nil)
```

Comment:
I recommend you to read https://pkg.go.dev/github.com/stretchr/testify/assert once. Many assertion funcs are available.

```suggestion
			if tc.wantErr {
				assert.Error(tt, err)
			} else {
				assert.NoError(tt, err)
			}
```